track/regen.go

```diff
@@ -0,0 +1,239 @@
+package track
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"regexp/syntax"
+)
+
+type CaptureGroupHandler func(index int, name string, group *syntax.Regexp, generator Generator, args *GeneratorArgs) string
+
+type generatorFactory func(regexp *syntax.Regexp, args *GeneratorArgs) (*internalGenerator, error)
+
+var generatorFactories map[syntax.Op]generatorFactory
+
+type GeneratorArgs struct {
+	Flags               syntax.Flags
+	CaptureGroupHandler CaptureGroupHandler
+}
+
+func (a *GeneratorArgs) initialize() error {
+	if a.CaptureGroupHandler == nil {
+		a.CaptureGroupHandler = defaultCaptureGroupHandler
+	}
+	return nil
+}
+
+// Generator generates random strings.
+type Generator interface {
+	Generate() string
+	String() string
+}
+
+func NewGenerator(pattern string, inputArgs *GeneratorArgs) (generator Generator, err error) {
```

Comment:
Any reason we're publicly exposing this func as well as the `Generator` interface? I don't